extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,417 @@
+    }
+    if (data) {
+      if (data.method === 'resize') {
+        /** Set the size of the card according to the message from Iframely */
+        const height = this.addBorderHeight_(me, data['height']);
+        me.attemptChangeHeight(height).catch(() => {});
+      }
+      if (data.method === 'setIframelyEmbedData') {
+        /** apply Iframely card styles if present */
+        const media = data['data']['media'] || null;
+        if (media && media['frame_style']) {
+          const styles = media['frame_style'].split(';');
+          styles.forEach(function(style) {
+            const props = style.split(':');
+            if (props.length === 2) {
+              const styleName = props[0].trim(),
+                styleValue = props[1].trim();
+              switch (styleName) {
+                case 'border':
+                  /** Because of the border-box sizing, changing border width doesn't change the size of the box. */
+                  setStyle(me.element, 'border', styleValue);
+                  /** Because Iframely sends border message before even building card's content, no change of iFrame's height is necessary here. */
+                  break;
+                case 'border-radius':
+                  setStyle(me.element, 'border-radius', styleValue);
+                  break;
+                case 'box-shadow':
+                  setStyle(me.element, 'box-shadow', styleValue);
+                  break;
+              }
+            }
+          });
+        }
+        if (media && media['aspect-ratio']) {
+          let height;
+          const box = me.element.getLayoutBox();
+          if (media['padding-bottom']) {
+            /** Apply height for media with updated "aspect-ratio" and "padding-bottom". */
+            height =
+              box.width / media['aspect-ratio'] + media['padding-bottom'];
+            height = this.addBorderHeight_(me, height);
+            me.attemptChangeHeight(height).catch(() => {});
+          } else {
+            height = box.width / media['aspect-ratio'];
+            if (Math.abs(box.height - height) > 1) {
+              /** Apply new height for updated "aspect-ratio". */
+              height = this.addBorderHeight_(me, height);
+              me.attemptChangeHeight(height).catch(() => {});
+            }
+          }
+        }
+      }
+      if (data.method === 'cancelWidget') {
+        me.attemptCollapse().catch(() => {});
+      }
+    }
+  }
+
+  /**
+   * Constructing placeholder image SRC
+   * @return {string} url of the placeholder
+   * @private
+   * */
+  constructPlaceholderSrc_() {
+    let src = null;
+    if (this.widgetId_) {
+      const url = this.base_ + this.widgetId_ + '/thumbnail';
+      src = addParamToUrl(url, 'amp', '1');
+    } else {
+      src = addParamsToUrl(this.base_ + 'api/thumbnail', {
+        'url': this.url_,
+        'key': this.key_,
+        'amp': '1',
+      });
+    }
+    return src;
```

Comment:
This is more complicated than it needs to be.

```suggestion
    if (this.widgetId_) {
      return `${this.base_}${this.widgetId_}/thumbnail?amp=1`;
    }
    return addParamsToUrl(`${this.base_}api/thumbnail`, {
      'url': this.url_,
      'key': this.key_,
      'amp': '1',
    });
```

If it's fine that `${widgetId}/thumbnail` gets superfluous params, also consider:

```suggestion
    return addParamsToUrl(
      `${this.base_}${this.widgetId_ || 'api'}/thumbnail`,
      {
        amp: 1,
        url: this.url_,
        key: this.key_,
      }
    );
```